src/pytorch_lightning/_graveyard/legacy_import_unpickler.py

```diff
@@ -1,20 +1,34 @@
 import pickle
 import warnings
-from typing import Any
+from typing import Any, Callable
 
+import torchmetrics
+from lightning_utilities.core.imports import compare_version as _compare_version
 
-class RedirectingUnpickler(pickle.Unpickler):
+
+def _patch_pl_to_mirror_if_necessary(module: str) -> str:
+    pl = "pytorch_" + "lightning"  # avoids replacement during mirror package generation
+    if module.startswith(pl):
+        # for the standalone package this won't do anything,
+        # for the unified mirror package it will redirect the imports
+        module = "pytorch_lightning" + module[len(pl) :]
+    return module
+
+
+class RedirectingUnpickler(pickle._Unpickler):
     def find_class(self, module: str, name: str) -> Any:
-        pl = "pytorch_" + "lightning"  # avoids replacement during mirror package generation
-        if module.startswith(pl):
-            # for the standalone package this won't do anything,
-            # for the unified mirror package it will redirect the imports
-            old_module = module
-            module = "pytorch_lightning" + module[len(pl) :]
-            # this warning won't trigger for standalone as these imports are identical
-            if module != old_module:
-                warnings.warn(f"Redirecting import of {old_module}.{name} to {module}.{name}")
-        return super().find_class(module, name)
+        new_module = _patch_pl_to_mirror_if_necessary(module)
+        # this warning won't trigger for standalone as these imports are identical
+        if module != new_module:
+            warnings.warn(f"Redirecting import of {module}.{name} to {new_module}.{name}")
+        return super().find_class(new_module, name)
+
+
+def compare_version(package: str, op: Callable, version: str, use_base_version: bool = False) -> bool:
+    new_package = _patch_pl_to_mirror_if_necessary(package)
+    return _compare_version(new_package, op, version, use_base_version)
 
 
+torchmetrics.utilities.imports._compare_version = compare_version  # type: ignore
+torchmetrics.metric._compare_version = compare_version  # type: ignore
```

Comment:
Maybe catch and ignore AttributeErrors? In case these paths get changed in the future

Also check `hasattr(compare_version)` to avoid patching something that doesn't exist

Reply:
done